cli/command/manifest/fetch.go

```diff
@@ -0,0 +1,205 @@
+package manifest
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/manifest/fetcher"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/registry"
+)
+
+// nolint: gocyclo
```

Comment:
It would be great if we could avoid adding any new functions to this gocyclo whitelist. I'm going to see if I can refactor this a bit and open a PR against this branch if I can find a way to split this up further.

I'm also wondering if this file could be moved to the `cli/manifest/` package, since it doesn't appear to define any commands.